File: src/Dto/SseSnapshot/TrainTaskSnapshot.ts
export interface TrainTaskSnapshot {
  /** 任务唯一标识 ID */
  taskId: number;
  /** 任务已执行时长（秒） */
  duration: number;
  /** 任务状态码（0:待处理, 1:运行中, 2:完成, 3:失败） */
  status: number;
  /** 任务状态描述信息 */
  statusMsg: string;
  /** 当前执行的轮次（epoch） */
  currentEpoch: number;
  /** 总训练轮次 */
  totalEpochs: number;
  /** 当前执行的批次（batch） */
  currentBatch: number;
  /** 单轮次总批次数量 */
  epochTotalBatches: number;
  /** 任务进度百分比（0.0 ~ 100.0） */
  progressPercent: number;
  /** 当前时间戳（格式：YYYY-MM-DD HH:mm:ss） */
  currentTime: string;
  /** 处理速度（可根据业务定义单位，如 batch/秒） */
  processSpeed: number;
  /** 模型准确率 */
  accuracy: number;
  /** 精确率 */
  precisionRate: number;
  /** 召回率 */
  recallRate: number;
  /** F1 分数（综合精确率和召回率的指标） */
  f1Score: number;
}

File: src/main/java/org/example/sentimentanalysis/dto/redisDto/TrainTaskSnapshot.java
package org.example.sentimentanalysis.dto.redisDto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class TrainTaskSnapshot {
    //     任务id
    private Long taskId;
    //    已持续时间
    private Integer duration;
    //  处理状态
    private Integer status;
    private String statusMsg;
    private Integer currentEpoch;
    //   当前时间
    // 关键点：匹配你的时间格式 "2026-02-13 20:17:02"
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime currentTime;
    //    平均处理速度
    private Integer totalEpochs;
    private Integer currentBatch;
    private Integer epochTotalBatches;
    private Double progressPercent;
    private String processSpeed;
//    性能指标
    private Double accuracy;
    private Double precisionRate;
    private Double recallRate;
    private Double f1Score;
}
